optimisation/design/Model_Optimisation_Design.md

```diff
@@ -473,14 +539,13 @@ bg = [OptFunction(@linear_in_h_bkg, hklT{:}, 'parameters', [1,0]), ...
 %	bg(3).binding == [@(p)-p(2), @(p)p(1)]
 %   bg(3).mapping = [2,4]
 
-
-opt = OptModel([sqw_object_1, sqw_object_2]);
-opt.set_multiplier_functions(multiplier);
-opt.set_foreground_functions(fg);
-opt.set_background_functions(bg);
+% Create the OptModel object
 % the first background function should be treated independently for each sqw
 % and all other functions apply to all sqw objects
-opt.set_background_applies({-1,0,0});
+opt = OptModel([sqw_object_1, sqw_object_2], ...
+               'foreground', [sw_func, ph_func],...
+               'background', OptFunctionsArray(bg, 'applies', {-1,0,0}), ...
+               'multiplier', multiplier);
```

Comment:
I guess here `OptModel` implicitly constructs a 2-element `OptFunctionsArray` for the input `[sw_func, ph_func]` and single-element `OptFunctionsArray` for the input `multiplier`?

Reply:
Implicit in this notation is that `OptModel` automatically constructs `OptFunctionsArray` objects from scalar or array `OptFunction` objects. The automatically constructed `OptFunctionsArray` objects would have default parameters (to be determined) so any non-default settings might be easiest accomplished by constructing the `OptFunctionsArray` object by hand, as in the case of the `'background'` functions.